R/render.R

```diff
@@ -325,6 +330,13 @@ render <- function(input,
       rmarkdown.runtime = runtime
     )
 
+    # read root directory from argument (has precedence) or front matter
+    root_dir <- knit_root_dir
```

Comment:
Could we support relative paths for knit_root_dir? I'm thinking especially of the YAML case for people that structure their projects with subdirectories for `data/`, `reports/`, etc.